### ISSUE TITLE
Replace fake Donated Items with real Donations

### ISSUE THREAD

**Issue by Anjali0407-git** [oss-slu/material-donor-mutual-assist#40]
Replace the hardcoded Donated Items list currently displayed in a tabular format with real, dynamic data. This will utilize data from the newly implemented Donated Item Form. Additionally, remove the unnecessary Barcode field and ensure all other fields remain visible. Update the donor field to show donor names (instead of just email) once the corresponding API is ready, and maintain existing filter and sorting features on the dynamic data.

**Current Situation:**

- The list of donations is currently displayed using hardcoded data, which limits its usefulness and relevance.
- A new Donated Item Form is now available that can be used to add new donations, making it possible to replace the hardcoded data with real dynamic data fetched from the backend.
- The Barcode field is no longer necessary, and we need to remove it from the table display.
- The current Donor Email field only shows an email string. Once the API is ready to provide all donor details, the Donor Name should be displayed instead of just the email.

Requirements:

- [x] **Replace Fake Data with Dynamic Data:**

- Populate the Donated Items table with real dynamic data fetched from the backend.
- Remove hardcoded data and replace it with real donations obtained via the appropriate API.

- [x] **Remove Barcode Field:**

- Remove the Barcode field from the displayed table, as it is no longer relevant.
- Ensure the remaining fields are still present and correctly displayed, including Item ID, Item Type, Current Status, Date Donated, Donor Email (or Donor Name), and Associated Program.
 

- [x] **Update Donor Field:**

For now, keep displaying the Donor Email in the table.
Once the API is updated to provide detailed donor information, update the table to show the Donor Name instead of just the email.
Ensure the frontend is set up to dynamically handle the updated data without requiring further changes to the logic.

- [x] **Ensure Filters and Sorting Work:**

Ensure the existing filters (e.g., by status, item type, program) continue to work seamlessly with the dynamic data.
Ensure the sorting functionality (e.g., sorting by date, donor, item type) still works as expected with the real donation data.

**Write Unit Tests:**
Write unit tests to verify that:

- The Donated Items table is correctly populated with real data from the backend.
- The Donor Name field is displayed correctly once donor details are provided via the updated API.
- The filter and sorting functionalities are functioning as expected with the new dynamic data.

Tests should cover scenarios such as:

- Successful Data Fetching: Ensure the data is successfully fetched from the backend and displayed in the table.
- Edge Cases: Verify that the table works correctly with empty data, partial data, and unexpected inputs.
- Filter and Sorting: Confirm that filtering and sorting features return accurate results based on the dynamic data.
